### PR TITLE
Add support to CMake to build swift-argument-parser during SwiftPM bootstrapping

### PR DESCRIPTION
**Pull request by owenv**
This is a fairly early attempt at integrating this repo into the SwiftPM bootstrapped build, someone better at CMake than I am should probably review :)

### CODE REVIEW

**Comment by natecook1000** [comment]
cc @compnerd ?

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,12 +1,23 @@
...
+
 project(swift-argument-parser
   LANGUAGES Swift)
 
 option(BUILD_EXAMPLES "Build Example Programs" TRUE)
+option(BUILD_SHARED_LIBS "Build shared libraries by default" YES)
 
 include(CTest)
 
+if(CMAKE_VERSION VERSION_LESS 3.16 AND CMAKE_SYSTEM_NAME STREQUAL Windows)
```
Comment (compnerd) [inline CMakeLists.txt:13]: Is this needed?  The minimum for this project is currently 3.16.  I think that we can ignore the <3.16 compatibility

File: Sources/CMakeLists.txt [hunk 2]
```diff
@@ -2,3 +2,9 @@ add_subdirectory(ArgumentParser)
 if(BUILD_TESTING)
   add_subdirectory(ArgumentParserTestHelpers)
 endif()
+
+set_property(GLOBAL APPEND PROPERTY ARGUMENTPARSER_EXPORTS ArgumentParser)
```
Comment (compnerd) [inline Sources/CMakeLists.txt:6]: Could use ArgumentParser_EXPORTS but, that’s a purely style thing, it has 0 bearing on anything.

File: CMakeLists.txt (hunk 1 above)
Comment (owenv) [reply CMakeLists.txt]: you're right, this was unnecessary, I dropped it

File: Sources/CMakeLists.txt (hunk 2 above)
Comment (owenv) [reply Sources/CMakeLists.txt]: Good to know, I'm not that familiar with CMake style conventions. I updated it since I was already making changes

**Comment by owenv** [comment]
@compnerd thanks for reviewing!

**Comment by natecook1000** [comment]
@swift-ci Please test

**Comment by compnerd** [comment]
@owenv my pleasure!

**Comment by natecook1000** [comment]
@swift-ci Please test

**Pull request merged**